### PR TITLE
py3-new (rebase): new testcase examples (using observer as example)

### PR DESCRIPTION
**Pull request by bernhardkaindl**
This is an example of adding a suite of tests that test a script (without mocks).

### CODE REVIEW

File: python3/packages/observer.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
 # We only want to import opentelemetry libraries if instrumentation is enabled
 # pylint: disable=import-outside-toplevel
 
-DEBUG_ENABLED = False
+DEBUG_ENABLED = os.getenv("OBSERVER_DEBUG")
```
Comment (psafont) [inline python3/packages/observer.py:44]: Another env variable is used for detecting the code is run in a test environment: `XAPI_TEST`

**Comment by psafont** [comment]
Aside from the name of the env variable, the PR loogks good to me

**Comment by gangj** [comment]
Hi @psafont @bernhardkaindl Thanks for letting me know this work!

Besides this PR, I see there are some python3 CI related changes in this feature branch: [xapi-project:feature/py3-new](https://github.com/xapi-project/xen-api/tree/feature/py3-new), would you please share some background of this work, and is there some relationship between this one and the one that Dao team is working on now([xapi-project:feature/py3](https://github.com/xapi-project/xen-api/tree/feature/py3)?

Thank you!

**Comment by psafont** [comment]
I was not aware of the branch `feature/py3-new`, I think this should be merged to `feature/py3`

**Comment by gangj** [comment]
Sorry, I see the PR for the commits in [xapi-project:feature/py3-new](https://github.com/xapi-project/xen-api/tree/feature/py3-new) just now: https://github.com/xapi-project/xen-api/pull/5504, if the change in this PR is irrelevant to the changes in the previous one, I think we should first work on and finish the previous one first? Or if they are related, then the changes here should be pushed into that PR and to be reviewed together?

**Comment by psafont** [comment]
Merge to a new python3 is not expected, let's close it and see if the code can be integrated in another way. I'll take a look at how to do this

**Pull request closed without merging**